assignments/prototypes.js

```diff
@@ -15,14 +15,38 @@
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
+function GameObject(gameObj) {
+  
+  this.dimensions = gameObj.dimensions;
+  this.createdAt = gameObj.createdAt;
+  this.name = gameObj.name; 
+
+}
+
+GameObject.prototype.destroy = function () {
+  return `${this.name} was removed from the game`;
+};
 
 /*
   === CharacterStats ===
   * healthPoints
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
+function CharacterStats(stats) {
+  this.hp = stats.hp;
```

Comment:
The object you passed in to `CharacterStats` should match the properties you are using. For example, the `mage` object doesn't have an `hp` property. It was called `healthPoints`. So logging `stats.hp` will give you `undefined`.